centreon/tests/php/www/class/CentreonDBTest.php

```diff
@@ -0,0 +1,493 @@
+<?php
+
+/*
+ * Copyright 2005 - 2024 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Tests\www\class;
+
+use CentreonDB;
+use CentreonDbConfig;
+use CentreonDbException;
+use CentreonDBStatement;
+use PDO;
+use PDOStatement;
+use ValueError;
+
+/**
+ * @param string $nameEnvVar
+ * @return string|null
+ */
+function getEnvironmentVariable(string $nameEnvVar): ?string
+{
+    $envVarValue = getenv($nameEnvVar, true) ?: getenv($nameEnvVar);
+    return (is_string($envVarValue) && !empty($envVarValue)) ? $envVarValue : null;
+}
+
+$dbHost = getEnvironmentVariable('MYSQL_HOST');
+$dbUser = getEnvironmentVariable('MYSQL_USER');
+$dbPassword = getEnvironmentVariable('MYSQL_PASSWORD');
+
+$dbConfig = null;
+
+if (!is_null($dbHost) && !is_null($dbUser) && !is_null($dbPassword)) {
+    $dbConfig = new CentreonDbConfig(
+        dbHostCentreon: $dbHost,
+        dbHostCentreonStorage: $dbHost,
+        dbUser: $dbUser,
+        dbPassword: $dbPassword,
+        dbNameCentreon: 'centreon',
+        dbNameCentreonStorage: 'centreon_storage',
+        dbPort: 3306
+    );
+}
+
+/**
+ * @param CentreonDbConfig $dbConfig
+ * @param string $dbName
+ * @return bool
+ */
+function hasConnectionDb(CentreonDbConfig $dbConfig, string $dbName): bool
+{
+    try {
+        if ($dbName === $dbConfig->getDbNameCentreon()) {
+            new PDO (
+                "mysql:dbname={$dbConfig->getDbNameCentreon()};host={$dbConfig->getDbHostCentreon()};port={$dbConfig->getDbPort()}",
+                $dbConfig->getDbUser(), $dbConfig->getDbPassword(), [PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION]
+            );
+        } elseif ($dbName === $dbConfig->getDbNameCentreonStorage()) {
+            new PDO (
+                "mysql:dbname={$dbConfig->getDbNameCentreonStorage()};host={$dbConfig->getDbHostCentreonStorage()};port={$dbConfig->getDbPort()}",
+                $dbConfig->getDbUser(), $dbConfig->getDbPassword(), [PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION]
+            );
+        } else {
+            return false;
+        }
```

Comment:
What about making CentreonDbConfig's responsibility to generate DSN string?
In order to have:
```suggestion
        new PDO ($dbConfig->getDsn($dbName));
```